tempodb/compactor_test.go

```diff
@@ -240,3 +240,77 @@ func TestSameIDCompaction(t *testing.T) {
 	}
 	assert.Equal(t, blockCount-blocksPerCompaction, records)
 }
+
+func TestCompactionUpdatesBlocklist(t *testing.T) {
+	tempDir, err := ioutil.TempDir("/tmp", "")
+	defer os.RemoveAll(tempDir)
+	assert.NoError(t, err, "unexpected error creating temp dir")
+
+	r, w, c, err := New(&Config{
+		Backend: "local",
+		Pool: &pool.Config{
+			MaxWorkers: 10,
+			QueueDepth: 100,
+		},
+		Local: &local.Config{
+			Path: path.Join(tempDir, "traces"),
+		},
+		WAL: &wal.Config{
+			Filepath:        path.Join(tempDir, "wal"),
+			IndexDownsample: rand.Int()%20 + 1,
+			BloomFP:         .01,
+		},
+		BlocklistPoll: 0,
+	}, log.NewNopLogger())
+	assert.NoError(t, err)
+
+	c.EnableCompaction(&CompactorConfig{
+		ChunkSizeBytes:          10,
+		MaxCompactionRange:      24 * time.Hour,
+		BlockRetention:          0,
+		CompactedBlockRetention: 0,
+	}, &mockSharder{})
+
+	// Cut x blocks with y records each
+	blockCount := 5
+	recordCount := 1
+	cutTestBlocks(t, w, blockCount, recordCount)
+
+	rw := r.(*readerWriter)
+	rw.pollBlocklist()
+
+	// compact everything
+	err = rw.compact(rw.blocklist(testTenantID), testTenantID)
+	assert.NoError(t, err)
+
+	// New blocklist contains 1 compacted block with everything
+	blocks := rw.blocklist(testTenantID)
+	assert.Equal(t, 1, len(blocks))
+	assert.Equal(t, uint8(1), blocks[0].CompactionLevel)
+	assert.Equal(t, blockCount*recordCount, blocks[0].TotalObjects)
+
```

Comment:
can we call `Find` here and confirm that tempodb can still find the expected traces?

Reply:
Added